src/spyglass/common/common_device.py

```diff
@@ -282,6 +285,30 @@ def insert_from_nwbfile(cls, nwbf):
                 }
                 cls.insert1(device_dict, skip_duplicates=True)
                 device_name_list.append(device_dict["camera_name"])
+        # Append devices from config file
+        if "CameraDevice" in config:
+            for conf in config["CameraDevice"]:
```

Comment:
For `for` loops like this, I advise declaring a `device_insert` list and then appending to it on each iteration. Then, `cls.insert(device_insert, skip_duplicates=True)` works the same, but it happens all at once protected in a db transaction - no partial inserts